src/core/proj/qgscoordinatereferencesystem.cpp

```diff
@@ -266,6 +267,29 @@ QgsCoordinateReferenceSystem QgsCoordinateReferenceSystem::createCompoundCrs( co
   return QgsCoordinateReferenceSystem();
 }
 
+QgsCoordinateReferenceSystem QgsCoordinateReferenceSystem::createGeocentricCrs( const QString &ellipsoid )
+{
+  const QgsEllipsoidUtils::EllipsoidParameters ellipsoidParams = QgsEllipsoidUtils::ellipsoidParameters( ellipsoid );
+  if ( !ellipsoidParams.valid )
+    return QgsCoordinateReferenceSystem();
+
+  QgsProjUtils::proj_pj_unique_ptr crs( proj_create_geocentric_crs(
+                                          QgsProjContext::get(),
+                                          /*crs_name*/ nullptr,
+                                          /*datum_name*/ nullptr,
```

Comment:
Not setting a datum will limit the purpose of the build geocentric CRS, given than any datum shift between it and other CRS will result in a ballpark (i.e. null) datum shift.
What is the intented use case for that method ?

Reply:
It's for the new 3d globe mode. The intention was that a geocentric crs matching the project's ellipsoid can be made and used for the globe. 

But alternatively, I could just drop this method and always use the other toGeocentricCrs method, starting from the project's crs...

Reply:
I've gone with adding a big warning to the method documentation:

"The returned CRS will be based on the ellipsoid parameters only, and will not have any datum associated with it. This prevents accurate datum shifts from occurring when transforming coordinates to or from the returned CRS. Any transformations involving this CRS will result in a ballpark (null) datum shift. The returned object should NOT be used when high accuracy transformations are required."